From ca1bb7b44f8b2b23fd5d14612693cfe3c2993290 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Tue, 22 Mar 2016 13:50:09 +0000
Subject: [PATCH] xfstests: strip out single quotes to enable fuzzier matching

Different core util versions emit single quotes or no single
quotes for utils such as mknod etc..  So strip these out before
we do the pattern matching comparisons.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 tests/generic/294     | 9 +++++++--
 tests/generic/294.out | 8 ++++----
 2 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/tests/generic/294 b/tests/generic/294
index 3fa6ba2..4e7a5aa 100755
--- a/tests/generic/294
+++ b/tests/generic/294
@@ -37,6 +37,11 @@ _cleanup()
     rm -f $tmp.*
 }

+_filter_single_quote()
+{
+    sed s/\'//g
+}
+
 # get standard environment, filters and checks
 . ./common/rc
 . ./common/filter
@@ -66,9 +71,9 @@ _scratch_mount
 rm -rf $THIS_TEST_DIR
 mkdir $THIS_TEST_DIR || _fail "Could not create dir for test"

-_create_files 2>&1 | _filter_scratch
+_create_files 2>&1 | _filter_single_quote | _filter_scratch
 _scratch_mount -o remount,ro || _fail "Could not remount scratch readonly"
-_create_files 2>&1 | _filter_scratch
+_create_files 2>&1 | _filter_single_quote | _filter_scratch

 # success, all done
 status=0
diff --git a/tests/generic/294.out b/tests/generic/294.out
index 1ac1c67..12e65ae 100644
--- a/tests/generic/294.out
+++ b/tests/generic/294.out
@@ -1,5 +1,5 @@
 QA output created by 294
-mknod: 'SCRATCH_MNT/294.test/testnode': File exists
-mkdir: cannot create directory 'SCRATCH_MNT/294.test/testdir': File exists
-touch: cannot touch 'SCRATCH_MNT/294.test/testtarget': Read-only file system
-ln: creating symbolic link 'SCRATCH_MNT/294.test/testlink': File exists
+mknod: SCRATCH_MNT/294.test/testnode: File exists
+mkdir: cannot create directory SCRATCH_MNT/294.test/testdir: File exists
+touch: cannot touch SCRATCH_MNT/294.test/testtarget: Read-only file system
+ln: creating symbolic link SCRATCH_MNT/294.test/testlink: File exists
--
2.7.3

